include/srf/segment/builder.hpp

```diff
@@ -95,6 +110,9 @@ class Builder final
     DELETE_COPYABILITY(Builder);
     DELETE_MOVEABILITY(Builder);
 
+    std::shared_ptr<ObjectProperties> get_ingress_dynamic(std::string name, std::type_index type_index);
+    std::shared_ptr<ObjectProperties> get_egress_dynamic(std::string name, std::type_index type_index);
+
```

Comment:
Could these just be an overload of `get_egress`? So you would have `get_egress(name)` and `get_egress(name, type_index)`. Ideally, the templated overload would just call the untemplated one passing `typeid(T)` for `type_index`

Reply:
Updated; I don't want to go so far as making the template version call the non-templated version, because they return different types and I don't know all the relevant use cases there; did switch it to an overload though.